src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/SupplementaryEvidenceCreator.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.SupplementaryEvidence;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.mappers.SupplementaryEvidenceMapper;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.model.Envelope;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+@Service
+public class SupplementaryEvidenceCreator {
+
+    private static final Logger log = LoggerFactory.getLogger(SupplementaryEvidenceCreator.class);
+
+    private static final String CASE_TYPE_ID = "Bulk_Scanned";
+    private static final String EVENT_TYPE_ID = "attachScannedDocs";
+
+    private final CcdAuthenticatorFactory authenticatorFactory;
+    private final CoreCaseDataApi coreCaseDataApi;
+
+    SupplementaryEvidenceCreator(
+        CcdAuthenticatorFactory authenticatorFactory,
+        CoreCaseDataApi coreCaseDataApi
+    ) {
+        this.authenticatorFactory = authenticatorFactory;
+        this.coreCaseDataApi = coreCaseDataApi;
+    }
+
+    public void createSupplementaryEvidence(Envelope envelope) {
+        log.info("Creating supplementary evidence for case {}", envelope.caseRef);
```

Comment:
since supplementary evidence processing is potentially going to master, this whole block should be skipped/erred in case `envelope.caseRef` is empty

Reply:
Will handle null

Reply:
Actually, I've just checked and as of now envelopes without case_ref won't even be parsed, as the field is required.